compass/compass.js

```diff
@@ -0,0 +1,53 @@
+﻿API.onUpdate.connect(function (sender) {
+
+    if (0.3 < API.getGameplayCamDir().X && 0.3 < API.getGameplayCamDir().Y) {

```

Comment:
There's absolutely no need to call `API.getGameplayCamDir()` 21 times. This is very slow.